plugins/network/NetClass.cpp

```diff
@@ -2193,16 +2200,66 @@ BOOL NetBrowser::IsReadable(const wchar_t* Remote)
 	return (FindHandle != INVALID_HANDLE_VALUE);
 }
 
-void NetBrowser::SetOpenFromCommandLine(wchar_t* ShareName)
+bool NetBrowser::SetOpenFromCommandLine(wchar_t* cmd)
 {
-	//lstrcpy (CmdLinePath, ShareName);
 #ifdef NETWORK_LOGGING
-	LogData(L"SetOpenFromCommandLine ShareName is");
-	LogData(ShareName);
+	LogData(L"SetOpenFromCommandLine cmd is");
+	LogData(cmd);
 #endif
-	lstrcpy(CmdLinePath, ShareName);
-	/*if(!GotoFavorite(ShareName))
-	  GotoComputer(ShareName);*/
+	int I = 0;
+	wchar_t* p = wcschr(cmd, L':');
+
+	if (!p || !*p)
+	{
+		return false;
+	}
+
+	*p++ = L'\0';
+	bool netg;
+
+	if (!lstrcmpi(cmd, L"netg"))
+		netg = true;
+	else if (!lstrcmpi(cmd, L"net"))
+		netg = false;
+	else
+	{
+		return false;
+	}
+
+	cmd = p;
+
+	if (lstrlen(FSF.Trim(cmd)))
+	{
+		if (cmd[0] == L'/')
+			cmd[0] = L'\\';
+
+		if (cmd[1] == L'/')
+			cmd[1] = L'\\';
+
+		if (!netg && !Opt.NavigateToDomains)
+		{
+			if (cmd[0] == L'\\' && cmd[1] != L'\\')
+				I = 1;
+			else if (cmd[0] != L'\\' && cmd[1] != L'\\')
+				I = 2;
+		}
+
+		wchar_t Path[MAX_PATH] = L"\\\\";
+		lstrcpy(Path + I, cmd);
+		FSF.Unquote(Path);
+		// Expanding environment variables.
+		{
+			wchar_t PathCopy[MAX_PATH];
+			lstrcpy(PathCopy, Path);
+			ExpandEnvironmentStrings(PathCopy, Path, static_cast<DWORD>(std::size(Path)));
+		}
+		lstrcpy(CmdLinePath, cmd);
+		if (!GotoFavorite(cmd))
```

Comment:
This is broken now.

[Previously](https://github.com/FarGroup/FarManager/pull/487/files#diff-f9560e1959a7a0320a77fd521da33f6fd4246ca31748fbe4094d12734f5f8187L119) SetOpenFromCommandLine was receiving the *already processed* (expanded, prefixed) path, storing it in CmdLinePath and passing it to GotoFavorite and GotoComputer.

Now the processed path (`Path`) is simply ignored and `net:localhost` doesn't work.

Reply:
https://github.com/FarGroup/FarManager/pull/491/commits/d5716bb914b4776b4d007dc1b427965ddf84b973